src/__tests__/actions/signupActions/__test__/signupActions.test.js

```diff
@@ -0,0 +1,67 @@
+import configureMockStore from "redux-mock-store";
+import thunk from "redux-thunk";
+import fetchMock from "fetch-mock";
+import signupAction from "../../../../actions/signupAction";
+import { REGISTER_SUCCESS, REGISTER_FAILURE } from "../../../../actions/types";
+
+const middlewares = [thunk];
+const mockStore = configureMockStore(middlewares);
+
+const { BASE_URL } = process.env;
+const signupUrl = `${BASE_URL}/users/`;
+
+describe("signup actions", () => {
+  afterEach(() => {
+    fetchMock.restore();
+  });
+
+  it("handles REGISTER_SUCCESS action after signing up", () => {
+    fetchMock.postOnce(signupUrl, {
+      body: { user: { token: "abdjakbd3jkbhdkkb3hjkbjdbjdjj" } },
+      headers: { "content-type": "application/json" },
+    });
+
+    const expectedAction = [{
+      type: REGISTER_SUCCESS,
+      payload: {
+        token: "abdjakbd3jkbhdkkb3hjkbjdbjdjj",
+        message: "you have successfully registered",
+      },
+    }];
+
+    const validData = {
+      username: "testuser1234",
+      email: "testuser1234@gmail.com",
+      password: "password1234",
+    };
+
+    const store = mockStore();
+
+    return store.dispatch(signupAction(validData)).then(() => {
+      expect(store.getActions()).toEqual(expectedAction);
+    });
+  });
+});
+
+it("handles REGISTER_FAILURE action after signing up", () => {
+  fetchMock.postOnce(signupUrl, {
+    body: { errors: {} },
+    headers: { "content-type": "application/json" },
+  });
+
+  const expectedAction = [{
+    type: REGISTER_FAILURE,
+    payload: {},
+  }];
+
+  const invalidData = {
+    email: "",
+    password: "",
+  };
+
+  const store = mockStore();
```

Comment:
You can create one const for `store` in this file in the describe

Reply:
I think it is better to have different stores so as to isolate the data between different tests.